src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQueryDatasetPdao.java

```diff
@@ -1428,4 +1430,35 @@ private static String bqStringValue(FieldValueList fieldValue, String fieldName)
     }
     return null;
   }
+
+  /// Snapshot Builder
+  // Pass in SQL query and map into generic BigQuery TableResult
+  public <T> List<T> runSnapshotBuilderQuery(
+      String sql, Dataset dataset, Function<TableResult, List<T>> aggregateSnapshotBuilderResults) {
```

Comment:
While this is currently used by snapshot builder, there isn't anything snapshot builder specific in this method. If so, should the name be changed to something more general?
```suggestion
  public <T> List<T> runQuery(
      String sql, Dataset dataset, Function<TableResult, List<T>> resultsMapper) {
```

Reply:
Yes, it can. I added a note that we expect for the string to be sanitized before getting passed into this method, in case it's utilized for more general purpose. Although, we might consider naming it something like "unsafeRunQuery" to make it really obvious.